File: Yandex/Practicum/Algorightms_and_Data_Structures/1_sprint/D.Хаотичность_погоды/solution.cpp
#include <iostream>
#include <vector>


void solution(int n, std::vector<int>& nums) {
    /*
    Time Complexity O(N)
    Memory Complexity O(1)
    */
    int output = 0;
    for (int index = 0; index < n; ++index) {
        if (
        (!index || nums[index - 1] < nums[index])
        && (index + 1 == n || nums[index] > nums[index + 1])
        ) {
            ++output;
        }
    }
    std::cout << output << "\n";
};


int main() {
    int n;
    std::cin >> n;
    std::vector<int> nums;
    for (int i = 0; i < n; ++i) {
        int num;
        std::cin >> num;
        nums.push_back(num);
    }
    solution(n, nums);
}

File: Yandex/Practicum/Algorightms_and_Data_Structures/1_sprint/D.Хаотичность_погоды/solution.py
import sys


def solution(n: int, nums: list[int]) -> None:
    """
    Time Complexity O(N)
    Memory Complexity O(1)
    """
    output: int = 0
    for idx in range(n):
        if (not idx or nums[idx - 1] < nums[idx]) and (idx + 1 == n or nums[idx] > nums[idx + 1]):
            output += 1
    sys.stdout.write(str(output) + '\n')


if __name__ == '__main__':
    n: int = int(sys.stdin.readline().rstrip())
    nums: list[int] = list(map(int, sys.stdin.readline().rstrip().split()))
    solution(n, nums)
